core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/AwsJsonProtocolErrorUnmarshaller.java

```diff
@@ -218,5 +237,17 @@ public Builder errorCodeParser(ErrorCodeParser errorCodeParser) {
         public AwsJsonProtocolErrorUnmarshaller build() {
             return new AwsJsonProtocolErrorUnmarshaller(this);
         }
+
+        /**
+         * Provides a check on whether AwsQueryCompatible trait is found in Metadata.
+         * If true, custom error codes can be provided
```

Comment:
Can we add more clarification like `Error code is derived from custom header` , or something like that specific to this.

Reply:
Sure. will add that.